src/main/java/com/exadel/discountwebapp/category/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.exadel.discountwebapp.category.service.impl;
+
+import com.exadel.discountwebapp.category.entity.Category;
+import com.exadel.discountwebapp.category.repository.CategoryRepository;
+import com.exadel.discountwebapp.category.service.CategoryService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CategoryServiceImpl implements CategoryService {
+
+    private final CategoryRepository categoryRepository;
+
+    @Autowired
+    public CategoryServiceImpl(CategoryRepository categoryRepository) {
+        this.categoryRepository = categoryRepository;
+    }
+
+    @Override
+    public List<Category> findAll() {
+        List<Category> categories = new ArrayList<>();
+        categoryRepository.findAll().forEach(categories::add);
+        return categories;
+    }
+
+    @Override
+    public Category findById(long id) {
+        return categoryRepository.findById(id).orElse(null);
+    }
+
+    @Override
+    public Category save(Category category) {
```

Comment:
better to name this method create instead of save. save can also mean "update" and thus confuse others

Reply:
renamed